### PR TITLE
Add coverage using nyc, badge to readme, and ignore the coverage outputs.

### PR DESCRIPTION
**Pull request by hildjj**
Fixes mafintosh/dns-packet#48

### CODE REVIEW

**Comment by hildjj** [comment]
As mentioned in the discussion of #48, requires these steps from @mafintosh :

You'll need a coveralls.io account (log in with github), and turn on support for this repo by clicking the "+" on the left side, searching for the repo name, and sliding the button to "on".

File: package.json [hunk 1]
```diff
@@ -10,7 +10,11 @@
     "node": ">=6"
   },
   "scripts": {
-    "test": "eslint --color *.js examples/*.js && tape test.js"
+    "clean": "rm -rf coverage .nyc_output/",
+    "lint": "eslint --color *.js examples/*.js",
+    "pretest": "npm run lint",
+    "test": "tape test.js",
```
Comment (silverwind) [inline package.json]: Can you include `lint` in the test command?

Comment (silverwind) [reply package.json]: Oh, I see it is in `pretest`. Is that guaranteed to execute on `test`?

**Comment by silverwind** [comment]
I was able to enable support for this repo on coveralls.io. Maybe push again here to see if it works?

File: package.json (hunk 1 above)
Comment (hildjj) [reply package.json]: Yes.  And is very slightly more portable.

**Comment by hildjj** [comment]
I think you may not see anything on coveralls until you merge this the first time.

File: .travis.yml [hunk 2]
```diff
@@ -1,4 +1,12 @@
 language: node_js
+sudo: false
```
Comment (silverwind) [inline .travis.yml]: What's this for? Maybe we want `dist: xenial`?

Comment (silverwind) [reply .travis.yml]: Thought, actually not a fan of specifying a OS version at all.

Comment (hildjj) [reply .travis.yml]: `sudo: false` allows them to use docker, which often leads to faster builds.  We don't need sudo support.  Apparently, that's now the default, so I can remove this.

Comment (hildjj) [reply .travis.yml]: After reading more, I see what you mean about `dist: xenial` (sorry for not reading deeper first!).  I can take any of these approaches you like:

1. leave `sudo: false`
2. remove it, and go back to all defaults
3. add `dist: xenial`

Comment (silverwind) [reply .travis.yml]: If `sudo: false` is the default, I vote for option 2.

Comment (hildjj) [reply .travis.yml]: I think `sudo: false` is only the default with `dist: xenial`, which I think you don't like.  I'll just remove it; worst case we're a tad slower.

**Pull request merged**